Packages/Netherlands3D/WebServiceReader/Runtime/Scripts/CreateWMSLayer.cs

```diff
@@ -50,135 +44,77 @@ public class WMSLOD
             new WMSLOD() { textureSize = 2048, maximumDistance = 1000 }
         };
 
-        [SerializeField]
-        private int tileSize = 1500;
+        [SerializeField] private int tileSize = 1500;
         [SerializeField] private bool compressLoadedTextures = true;
-        private bool OnlyOnTerrain_Memory = false;
         private bool DisplayState = true;
 
         void Start()
         {
-            tileHandler = FindObjectOfType(typeof(TileSystem.TileHandler)) as TileSystem.TileHandler;
-            if (OnWMSUrlDefined_String)
-            {
-                OnWMSUrlDefined_String.AddListenerStarted(CreateWMS_Layer);
-            }
-            if (ShowWMSOnBuildings)
-            {
-                ShowWMSOnBuildings.AddListenerStarted(showWMSOnBuildings);
-            }
-            if (ShowWMSOnTerrain)
-            {
-                ShowWMSOnTerrain.AddListenerStarted(showWMSOnTerrain);
-            }
-            if (ShowWMSOnBuildingsAndTerrain)
-            {
-                ShowWMSOnBuildingsAndTerrain.AddListenerStarted(showWMSOnBuildingsAndTerrain);
-            }
-            if (UnloadWMSService)
-            {
-                UnloadWMSService.AddListenerStarted(UnloadLayer);
-            }
-            if (AleenOpMaaiveld_Bool)
-            {
-                AleenOpMaaiveld_Bool.AddListenerStarted(ShowOnlyOnTerrain);
-            }
-            if (ShowLayer_Bool)
-            {
-                ShowLayer_Bool.AddListenerStarted(ShowLayer);
-            }
+            tileHandler = FindObjectOfType(typeof(TileHandler)) as TileHandler;
+            if (!tileHandler)
+                Debug.LogWarning("No TileHandler found. This script depends on a TileHandler.", this.gameObject);
         }
+
         /// <summary>
         /// turn the layer on or off
         /// </summary>
         /// <param name="OnOff"></param>
         public void ShowLayer(bool OnOff)
         {
             DisplayState = OnOff;
+            Debug.Log("Show layer " + OnOff);
```

Comment:
Zijn deze debug statement en die op regel 78 bewust erin gelaten?

Reply:
Ja wel bewust idd. Persoonlijk vind ik dit soort log meldingen in de console wel handig zodat ik zie of een bepaalde methode reageert op een actie die ik doe, los van de verwachte visuele uitkomst ( in dit geval verschijnen en verdwijnen  van de projectie laag. )
Bijv: Mocht mijn laag niet verschijnen wanneer ik op een knop druk maar wel de melding in de console zie weet ik dat ik het probleem aan de renderlagen kant moet gaan zoeken. 
Zolang het geen console spam wordt vind ik dit soort logs handig. Wat vind jij? We zouden ook evt. altijd kunnen eisen dat het 'loggen' van een systeem altijd met een bool aan of uitgezet kan worden, naast de bestaande mogellijkheid om alle logs uit te zetten met bijv. Debug.unityLogger.logEnabled = Debug.isDebugBuild;

Reply:
dit nu omgezet naar losse events + voorbeeld logger in Core